### ISSUE TITLE
Properly handle workspace deletions

### ISSUE THREAD

**Issue by jayrbolton** [kbase/index_runner_deluxe#88]
I thought we were handling this, but here are the logs:

```
1/2/2020 1:11:13 PMINFO:root:es_writer received WORKSPACE_STATE_CHANGE for 45505/None
1/2/2020 1:11:13 PMINFO:root:Unrecognized event WORKSPACE_STATE_CHANGE.
1/2/2020 1:11:13 PMINFO:root:End of stream.
1/2/2020 1:11:13 PMINFO:root:Received WORKSPACE_STATE_CHANGE for 45505/None
1/2/2020 1:11:13 PMINFO:root:Unrecognized event WORKSPACE_STATE_CHANGE.
1/2/2020 1:13:38 PMINFO:root:End of stream.
1/2/2020 1:13:38 PMINFO:root:Received WORKSPACE_DELETE_STATE_CHANGE for 45505/None
1/2/2020 1:13:38 PMINFO:root:es_writer received WORKSPACE_DELETE_STATE_CHANGE for 45505/None
1/2/2020 1:13:38 PMINFO:root:_delete_ws TODO
1/2/2020 1:13:38 PMERROR:root:================================================================================
1/2/2020 1:13:38 PMWorkspace error with code 500:
1/2/2020 1:13:38 PM{"version":"1.1","error":{"name":"JSONRPCError","code":-32500,"message":"Workspace 45505 is deleted","error":"us.kbase.workspace.database.exceptions.NoSuchWorkspaceException: Workspace 45505 is deleted\n\tat us.kbase.workspace.database.mongo.MongoWorkspaceDB.resolveWorkspaces(MongoWorkspaceDB.java:1035)\n\tat us.kbase.workspace.database.mongo.MongoWorkspaceDB.resolveWorkspaces(MongoWorkspaceDB.java:1010)\n\tat us.kbase.workspace.database.mongo.MongoWorkspaceDB.resolveWorkspaces(MongoWorkspaceDB.java:998)\n\tat us.kbase.workspace.database.Workspace.listObjects(Workspace.java:1019)\n\tat us.kbase.workspace.kbase.WorkspaceServerMethods.listObjects(WorkspaceServerMethods.java:600)\n\tat us.kbase.workspace.kbase.admin.WorkspaceAdministration.runCommand(WorkspaceAdministration.java:369)\n\tat us.kbase.workspace.WorkspaceServer.administer(WorkspaceServer.java:1725)\n\tat sun.reflect.GeneratedMethodAccessor137.invoke(Unknown Source)\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n\tat java.lang.reflect.Method.invoke(Method.java:498)\n\tat us.kbase.common.service.JsonServerServlet.processRpcCall(JsonServerServlet.java:610)\n\tat us.kbase.common.service.JsonServerServlet.doPost(JsonServerServlet.java:463)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:661)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:742)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)\n\tat org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)\n\tat org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)\n\tat org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)\n\tat org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:470)\n\tat org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140)\n\tat org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:80)\n\tat org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:624)\n\tat org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)\n\tat org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342)\n\tat org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:799)\n\tat org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)\n\tat org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:861)\n\tat org.apache.tomcat.util.net.Nio2Endpoint$SocketProcessor.doRun(Nio2Endpoint.java:1698)\n\tat org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)\n\tat org.apache.tomcat.util.net.AbstractEndpoint.processSocket(AbstractEndpoint.java:934)\n\tat org.apache.tomcat.util.net.Nio2Endpoint$Nio2SocketWrapper$4.completed(Nio2Endpoint.java:627)\n\tat org.apache.tomcat.util.net.Nio2Endpoint$Nio2SocketWrapper$4.completed(Nio2Endpoint.java:605)\n\tat sun.nio.ch.Invoker.invokeUnchecked(Invoker.java:126)\n\tat sun.nio.ch.Invoker$2.run(Invoker.java:218)\n\tat sun.nio.ch.AsynchronousChannelGroupImpl$1.run(AsynchronousChannelGroupImpl.java:112)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\n"}}
1/2/2020 1:13:38 PM--------------------------------------------------------------------------------
1/2/2020 1:13:38 PMTraceback (most recent call last):
1/2/2020 1:13:38 PM  File "/app/src/utils/worker_group.py", line 91, in _run_worker
1/2/2020 1:13:38 PM    inst.on_queue_empty()
1/2/2020 1:13:38 PM  File "/app/src/index_runner/es_writer.py", line 77, in on_queue_empty
1/2/2020 1:13:38 PM    self._perform_batch_deletes(min_length=1)
1/2/2020 1:13:38 PM  File "/app/src/index_runner/es_writer.py", line 114, in _perform_batch_deletes
1/2/2020 1:13:38 PM    _delete_from_elastic(self.batch_deletes)
1/2/2020 1:13:38 PM  File "/app/src/index_runner/es_writer.py", line 234, in _delete_from_elastic
1/2/2020 1:13:38 PM    for (obj_id, ver) in _WS_CLIENT.generate_all_ids_for_workspace(wsid, admin=True):
1/2/2020 1:13:38 PM  File "/usr/local/lib/python3.7/site-packages/kbase_workspace_client/main.py", line 91, in generate_all_ids_for_workspace
1/2/2020 1:13:38 PM    part = self.admin_req("listObjects", params)
1/2/2020 1:13:38 PM  File "/usr/local/lib/python3.7/site-packages/kbase_workspace_client/main.py", line 114, in admin_req
1/2/2020 1:13:38 PM    return _post_req(payload, self._ws_url, self._token)
1/2/2020 1:13:38 PM  File "/usr/local/lib/python3.7/site-packages/kbase_workspace_client/main.py", line 21, in _post_req
1/2/2020 1:13:38 PM    raise WorkspaceResponseError(resp)
1/2/2020 1:13:38 PMkbase_workspace_client.exceptions.WorkspaceResponseError: Workspace error with code 500:
1/2/2020 1:13:38 PM{"version":"1.1","error":{"name":"JSONRPCError","code":-32500,"message":"Workspace 45505 is deleted","error":"us.kbase.workspace.database.exceptions.NoSuchWorkspaceException: Workspace 45505 is deleted\n\tat us.kbase.workspace.database.mongo.MongoWorkspaceDB.resolveWorkspaces(MongoWorkspaceDB.java:1035)\n\tat us.kbase.workspace.database.mongo.MongoWorkspaceDB.resolveWorkspaces(MongoWorkspaceDB.java:1010)\n\tat us.kbase.workspace.database.mongo.MongoWorkspaceDB.resolveWorkspaces(MongoWorkspaceDB.java:998)\n\tat us.kbase.workspace.database.Workspace.listObjects(Workspace.java:1019)\n\tat us.kbase.workspace.kbase.WorkspaceServerMethods.listObjects(WorkspaceServerMethods.java:600)\n\tat us.kbase.workspace.kbase.admin.WorkspaceAdministration.runCommand(WorkspaceAdministration.java:369)\n\tat us.kbase.workspace.WorkspaceServer.administer(WorkspaceServer.java:1725)\n\tat sun.reflect.GeneratedMethodAccessor137.invoke(Unknown Source)\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n\tat java.lang.reflect.Method.invoke(Method.java:498)\n\tat us.kbase.common.service.JsonServerServlet.processRpcCall(JsonServerServlet.java:610)\n\tat us.kbase.common.service.JsonServerServlet.doPost(JsonServerServlet.java:463)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:661)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:742)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)\n\tat org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)\n\tat org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)\n\tat org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)\n\tat org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)\n\tat org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)\n\tat org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:470)\n\tat org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140)\n\tat org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:80)\n\tat org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:624)\n\tat org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)\n\tat org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342)\n\tat org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:799)\n\tat org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)\n\tat org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:861)\n\tat org.apache.tomcat.util.net.Nio2Endpoint$SocketProcessor.doRun(Nio2Endpoint.java:1698)\n\tat org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)\n\tat org.apache.tomcat.util.net.AbstractEndpoint.processSocket(AbstractEndpoint.java:934)\n\tat org.apache.tomcat.util.net.Nio2Endpoint$Nio2SocketWrapper$4.completed(Nio2Endpoint.java:627)\n\tat org.apache.tomcat.util.net.Nio2Endpoint$Nio2SocketWrapper$4.completed(Nio2Endpoint.java:605)\n\tat sun.nio.ch.Invoker.invokeUnchecked(Invoker.java:126)\n\tat sun.nio.ch.Invoker$2.run(Invoker.java:218)\n\tat sun.nio.ch.AsynchronousChannelGroupImpl$1.run(AsynchronousChannelGroupImpl.java:112)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\n"}}
1/2/2020 1:13:38 PM
1/2/2020 1:13:38 PM================================================================================
```

I don't see any indication that we are really handling this well.

**Comment by jayrbolton** [comment]
Origin of the bug is here: https://github.com/kbase/index_runner_deluxe/blob/develop/src/index_runner/es_writer.py#L233

We are trying to fetch the workspace, but it is deleted.

Instead of fetching all the exact object ids under the workspace, I imagine we could perform a bulk delete by querying for anything that has the ID `WS::{wsid}*`

**Comment by jayrbolton** [comment]
Fix in PR #93
